core/domain/wipeout_service.py

```diff
@@ -339,18 +347,160 @@ def _pseudonymize_models(activity_related_models, pseudonymized_user_id):
+            request object to be saved in the datastore.
     """
-    for model_class in models.Registry.get_storage_model_classes(
-            [models.NAMES.user]):
-        if (model_class.get_deletion_policy() not in
-                [base_models.DELETION_POLICY.KEEP,
-                 base_models.DELETION_POLICY.NOT_APPLICABLE]):
-            model_class.apply_deletion_policy(user_id)
+    user_id = pending_deletion_request.user_id
+
+    feedback_thread_model_class = feedback_models.GeneralFeedbackThreadModel
+    feedback_thread_models = feedback_thread_model_class.query(ndb.OR(
+        feedback_thread_model_class.original_author_id == user_id,
+        feedback_thread_model_class.last_nonempty_message_author_id == user_id
+    )).fetch()
+    feedback_ids = set([model.id for model in feedback_thread_models])
+
+    feedback_message_model_class = feedback_models.GeneralFeedbackMessageModel
+    feedback_message_models = feedback_message_model_class.query(
+        feedback_message_model_class.author_id == user_id
+    ).fetch()
+    feedback_ids |= set([model.thread_id for model in feedback_message_models])
+
+    suggestion_model_class = suggestion_models.GeneralSuggestionModel
+    general_suggestion_models = suggestion_model_class.query(ndb.OR(
+        suggestion_model_class.author_id == user_id,
+        suggestion_model_class.final_reviewer_id == user_id
+    )).fetch()
+    feedback_ids |= set([model.id for model in general_suggestion_models])
+
+    # The activity_mappings field might have only been partially generated, so
+    # we fill in the missing part for this activity category.
+    if models.NAMES.feedback not in pending_deletion_request.activity_mappings:
+        pending_deletion_request.activity_mappings[models.NAMES.feedback] = (
+            _generate_activity_to_pseudonymized_ids_mapping(feedback_ids)
+        )
+        save_pending_deletion_request(pending_deletion_request)
+
+    def _pseudonymize_models(feedback_related_models, pseudonymized_user_id):
+        """Pseudonymize user ID fields in the models.
+
+        Args:
+            feedback_related_models: list(BaseModel). Models whose user IDs
+                should be pseudonymized.
+            pseudonymized_user_id: str. New pseudonymized user ID to be used for
+                the models.
+        """
+        feedback_thread_models = [
+            model for model in feedback_related_models
+            if isinstance(model, feedback_thread_model_class)]
+        for feedback_thread_model in feedback_thread_models:
+            if feedback_thread_model.original_author_id == user_id:
+                feedback_thread_model.original_author_id = pseudonymized_user_id
+            if feedback_thread_model.last_nonempty_message_author_id == user_id:
+                feedback_thread_model.last_nonempty_message_author_id = (
+                    pseudonymized_user_id)
+        feedback_thread_model_class.put_multi(feedback_thread_models)
```

Comment:
Should we do a ndb.put_multi() for all three types of models at once? I'm not sure.

Reply:
Yep, make sense, done (also fixed in the `_pseudonymize_activity_models`).